Fix the bug with ignoring -B option. http://llvm.org/bugs/show_bug.cgi?id=10744
Check if it is already resolved when you advance Clang revision.

--- a/asan/clang_src/tools/clang/lib/Driver/ToolChains.cpp
+++ b/asan/clang_src/tools/clang/lib/Driver/ToolChains.cpp
@@ -1621,11 +1621,12 @@ Linux::Linux(const HostInfo &Host, const llvm::Triple &Triple)
     Lib = Lib64;
   }
 
-  llvm::sys::Path LinkerPath(Base + "/../../../../" + GccTriple + "/bin/ld");
-  if (!llvm::sys::fs::exists(LinkerPath.str(), Exists) && Exists)
-    Linker = LinkerPath.str();
-  else
-    Linker = GetProgramPath("ld");
+  // OpenSuse stores the linker with the compiler, add that to the search
+  // path.
+  ToolChain::path_list &PPaths = getProgramPaths();
+  PPaths.push_back(Base + "/../../../../" + GccTriple + "/bin");
+
+  Linker = GetProgramPath("ld");
 
   LinuxDistro Distro = DetectLinuxDistro(Arch);
 
